### PR TITLE
feat: ora staff grader backend

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** ORA Staff Grader Backend

JIRA 
- Merge Ticket [AU-508](https://openedx.atlassian.net/browse/AU-508)
- Backend Epic [AU-49](https://openedx.atlassian.net/browse/AU-49)

**What changed?**

- Add staffgrader app
- Add staff_grader_mixin which adds several xblock handlers that act as a json ora api for listing submissions and getting assessment lock and submission information to enable the frontend.
- Add SubmissionGradingLock model to represent a "locked" submission so that only one grader may grade a submission at a given time.
- Add create_oa_submissions_from_file management command for bulk creation of test users submissions assessments and locks
- Add ESG linking form instructor dashboard
- Add file size info to file descriptors
- Remove debug toolbar from ORA

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Comment by jansenk** [comment]
- [ ] Bump version number

File: openassessment/staffgrader/serializers/assessments.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+
+class AssessmentPartSerializer(serializers.ModelSerializer):
+    name = serializers.CharField(source='criterion.name')
+    option = serializers.CharField(source='option.name', default=None)
+    points = serializers.IntegerField(source='option.points', default=None)
+
+    class Meta:
+        model = AssessmentPart
+        fields = ['name', 'option', 'points', 'feedback']
```
Comment (nsprenkle) [inline openassessment/staffgrader/serializers/assessments.py:27]: Where does 'feedback' come from? Same question for `AssessmentScoreSerializer`.

File: openassessment/staffgrader/tests/test_base.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+class StaffGraderMixinTestBase(XBlockHandlerTestCase):
+    """ Base class for esg-related tests, with common setup and utility methods """
+
+    handler_name = None
+
+    def setUp(self):
+        super().setUp()
+        # Lots of large dict comparisons in this file so display full diff
+        self.maxDiff = None
```
Comment (nsprenkle) [inline openassessment/staffgrader/tests/test_base.py:19]: That's cool, didn't know you could do that.

**Comment by jansenk** [comment]
Opened https://github.com/openedx/edx-ora2/pull/1796 to address quality failures that somehow slipped in.

File: openassessment/xblock/openassessmentblock.py [hunk 3]
```diff
@@ -586,7 +588,8 @@ def ora_blocks_listing_view(self, context=None):
             additional_css=["static/css/lib/backgrid/backgrid%s.css" % min_postfix],
             additional_js=["static/js/lib/backgrid/backgrid%s.js" % min_postfix],
             additional_js_context={
-                "ENHANCED_STAFF_GRADER": self.is_enhanced_staff_grader_enabled
+                "ENHANCED_STAFF_GRADER": self.is_enhanced_staff_grader_enabled,
+                "ORA_GRADING_MICROFRONTEND_URL": settings.ORA_GRADING_MICROFRONTEND_URL
```
Comment (nsprenkle) [inline openassessment/xblock/openassessmentblock.py:592]: Do we need handling for when this config isn't present?

**Review by nsprenkle — COMMENTED** [review]
Some tidying jobs that need to take place: refresh translations, build JS, version bump.

Should also add missing changes to PR description for visibility: adds ESG linking form instructor dashboard, adds file size info to file descriptors, removes debug toolbar from ORA.

**Comment by nsprenkle** [comment]
Annoyingly, it looks like pylint got updated underneath us and there are new linting rules.

File: openassessment/staffgrader/serializers/assessments.py (hunk 1 above)
Comment (jansenk) [reply openassessment/staffgrader/serializers/assessments.py:27]: In both places they come directly from the model. If you have a field on a model that's specify in fields on a ModelSerializer it will automatically include it. The other fields have to be manually defined because they're doing `source`-y stuff

Comment (jansenk) [reply openassessment/staffgrader/serializers/assessments.py:27]: And I assume you mean `AssessmentSerializer`

File: openassessment/xblock/openassessmentblock.py (hunk 3 above)
Comment (jansenk) [reply openassessment/xblock/openassessmentblock.py:592]: Yeah, I'd agree and say this should maybe be`getattr(settings, 'ORA_GRADING_MICROFRONTEND_URL', '')`

Comment (jansenk) [reply openassessment/xblock/openassessmentblock.py:592]: Just to check I threw together this PR(which includes an old commit too, my bad) and pylint doesn't complain. Are you sure we needed that ignore? https://github.com/openedx/edx-ora2/pull/1797

**Review by nsprenkle — APPROVED** [review]
LGTM, with obligatory reminder to do the following before merge:
- [x] Rerun translations
- [x] Build static assets
- [ ] Bump version number

**Comment by jansenk** [comment]
Hmm, codecov is complaining about the fact that my management command isn't tested. I could throw some tests together or we could just ignore the failure and handle it later. @nsprenkle what do you think? I'd personally like to get thus merged before I'm gone for my long weekend.
